p2p/module.go

```diff
@@ -125,7 +107,7 @@ func (m *p2pModule) Start() error {
 		return err
 	}
 
-	if m.p2pConfig.GetUseRainTree() {
+	if m.p2pConfig.(*typesP2P.P2PConfig).UseRainTree {
```

Comment:
I feel like we can locally store the casted type on line 31.

Reply:
Addressed in https://github.com/pokt-network/pocket/pull/235/commits/c60206fc430c0b2a8c93ec473d9626aaebadef20

Also applied the same pattern to the other modules for consistency and also removed config related fields from the modules to use the config instead.